src/builder.rs

```diff
@@ -139,6 +152,14 @@ pub enum BuildError {
     DuplicateSignature,
     /// The bundle being constructed violated the construction rules for the requested bundle type.
     BundleTypeNotSatisfiable,
+    /// Native asset cannot be burned
+    BurnNative,
+    /// The value to be burned cannot be zero
+    BurnZero,
+    /// The asset to be burned is duplicated.
+    BurnDuplicateAsset,
+    /// There is no available split note for this asset.
+    NoSplitNoteAvailable,
```

Comment:
- Instead of having three separate `Burn*` variants that directly mirror `BurnError` variants, have a single `Burn(BurnError)` variant.
- Document the resulting addition in the changelog. (Previously they would have been changes, but after #473 this error enum is non-exhaustive.)